test/distributed/org/apache/cassandra/fuzz/topology/HarryTopologyMixupTest.java

```diff
@@ -66,65 +107,130 @@ protected void destroyState(State<Spec> state, @Nullable Throwable cause)
             TopologyHistory history = state.topologyHistory;
             // if any topology change happened, then always validate all
             if (harryState.generation != history.generation())
             {
                 harryState.generation = history.generation();
-                return validateAll;
+                return validateAll(state);
             }
             if ((harryState.numInserts > 0 && rs.decide(0.2))) // 20% of the time do reads
-                return validateAll;
+                return validateAll(state);
             return insert;
         };
     }
 
+    private static Command<State<Spec>, Void, ?> validateAll(State<Spec> state)
+    {
+        Spec spec = state.schemaSpec;
+        var schema = spec.harry.schema();
+        boolean writeThroughAccord = schema.isWriteTimeFromAccord();
+        List<Command<State<Spec>, Void, ?>> reads = new ArrayList<>();
+        Model model = spec.harry.quiescentChecker();
+        for (Long pd : new TreeSet<>(spec.harry.pds()))
+        {
+            reads.add(new HarryCommand(s -> "Harry Validate pd=" + pd  + state.commandNamePostfix(), s -> model.validate(Query.selectAllColumns(schema, pd, false))));
+            // as of this writing Accord does not support ORDER BY
+            if (!writeThroughAccord)
+                reads.add(new HarryCommand(s -> "Harry Reverse Validate pd=" + pd + state.commandNamePostfix(), s -> model.validate(Query.selectAllColumns(schema, pd, true))));
+        }
+//        if (reads.isEmpty())
+//            throw new IllegalStateException("Attempted to read when no partitions have been written to");
+        reads.add(new HarryCommand(s -> "Reset Harry Write State" + state.commandNamePostfix(), s -> ((HarryState) s).numInserts = 0));
+        return Property.multistep(reads);
+    }
+
+    private static class AccordSut extends InJvmSut
+    {
+        private AccordSut(Cluster cluster)
+        {
+            super(cluster, roundRobin(cluster), retryOnTimeout(), 10, 3);
+        }
+
+        @Override
+        public Object[][] execute(String statement, ConsistencyLevel cl, int coordinator, int pageSize, Object... bindings)
+        {
+            return super.execute(wrapInTxn(statement), cl, coordinator, pageSize, bindings);
+        }
+
+        @Override
+        protected void onException(Throwable t)
+        {
+            t = Throwables.getRootCause(t);
+            if (!TIMEOUT_CHECKER.matches(t)) return;
+
+            TxnId id;
+            try
+            {
+                id = TxnId.parse(t.getMessage());
+            }
+            catch (Throwable t2)
+            {
+                return;
+            }
+            try
+            {
+                var nodes = cluster.stream().filter(i -> !i.isShutdown()).mapToInt(i -> i.config().num()).toArray();
+                logger.warn("Timeout for txn {}; debug info\n{}", id, ClusterUtils.queryTxnStateAsString(cluster, id, nodes));
```

Comment:
when harry sees a timeout, try to quickly get the cluster state to see *why* it was timed out... this queries a vtable